### PR TITLE
[precommit-setx] Extending no set -x precommit to all OSes

### PR DESCRIPTION
**Pull request by CelianR**
### What does this PR do?

The `shell-check-no-set-x` pre-commit hook wasn't portable for all oses, used python for portable code.

### CODE REVIEW

File: tasks/pre_commit.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from invoke import task
+from invoke.context import Context
+
+from tasks.libs.common.color import color_message
+
+
+@task()
```
Comment (amenasria) [inline tasks/pre_commit.py:7]:
```suggestion
@task
```

File: tasks/pre_commit.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+def shell_check_no_set_x(ctx: Context):
+    """
+    Check that shell scripts do not use 'set -x'"
```
Comment (amenasria) [inline tasks/pre_commit.py:10]: **💬 suggestion:**  Could you please add the xtrace one as well ?

File: tasks/pre_commit.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    """
+    command = "git grep -rnE --color=always 'set( +-[^ ])* +-[^ ]*(x|( +xtrace))' -- ':*.sh' ':*/Dockerfile' ':*.yaml' ':*.yml' ':(exclude).pre-commit-config.yaml'"
+
+    result = ctx.run(command, hide=True, warn=True)
+    if result.return_code == 0:
+        print(result.stdout.strip())
+        print()
```
Comment (amenasria) [inline tasks/pre_commit.py:17]:
```suggestion
        print(result.stdout.strip(), end="\n\n")
```

File: tasks/pre_commit.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        print(color_message('error:', 'red'), 'No shell script should use "set -x"')
```
Comment (amenasria) [inline tasks/pre_commit.py:18]:
```suggestion
        print(color_message('error:', 'red'), 'No shell script should use "set -x"', file=sys.stderr)
```
(with an `import sys` at the start)

**Review by amenasria — APPROVED** [review]
LGTM ! Left a few comments !


**👏 praise:** ‏Bangerrrr

**Review by chouetz — APPROVED** [review]

**Review by FlorentClarret — APPROVED** [review]

**Pull request merged**